components/dashboard-modules/EnergyReccomendation.tsx

```diff
@@ -0,0 +1,71 @@
+import Image from 'next/image';
+import { FC } from 'react';
+import { useSiteData } from '~/lib/hooks';
+import { getCurrentTimeForecast } from '~/lib/utils';
+import content from '../../content/power-card-content.json';
+import NumberDisplay from './NumberDisplay';
+
+/**
+ * Determines the appliance with the greatest energy required that is less than or equal to the current output
+ * @param currentOutput the current output in kW based on the getCurrentTimeForecast
+ * @returns the index of the appliance with the greatest enrgy requiement less than or equal to the current ouput
+ * or -1 if an index could not be found
+ */
+
+const getBestRecomendationIndex = (currentOutput: number | undefined) => {
+  if (currentOutput != undefined) {
+    let maxIndex = -1;
```

Comment:
let's have this be initialized to `undefined`, and have the function return `undefined` when there doesn't exist a suitable index

Reply:
I just don't really like having sentinel numbers here like `-1`; if we don't have a match, returning `undefined` (or even `null`, `null` might be better) seems more honest